.azure/scripts/setup-kind.sh

```diff
@@ -142,27 +230,28 @@ if [[ "$IP_FAMILY" = "ipv4" || "$IP_FAMILY" = "dual" ]]; then
     # https://github.com/kubernetes-sigs/kind/issues/2875
     # https://github.com/containerd/containerd/blob/main/docs/cri/config.md#registry-configuration
     # See: https://github.com/containerd/containerd/blob/main/docs/hosts.md
-    cat <<EOF | kind create cluster --image "${KIND_NODE_IMAGE}" --config=-
-    kind: Cluster
-    apiVersion: kind.x-k8s.io/v1alpha4
-    nodes:
-    - role: control-plane
-    - role: control-plane
-    - role: control-plane
-    - role: worker
-    - role: worker
-    - role: worker
-    name: $KIND_CLUSTER_NAME
-    containerdConfigPatches:
-    - |-
-     [plugins."io.containerd.grpc.v1.cri".registry]
-       config_path = "/etc/containerd/certs.d"
-    networking:
-       ipFamily: $IP_FAMILY
+    cat <<EOF | sudo systemd-run --scope -p "Delegate=yes" /usr/local/bin/kind create cluster --image "${KIND_NODE_IMAGE}" --name "${KIND_CLUSTER_NAME}" --config=-
+        kind: Cluster
+        apiVersion: kind.x-k8s.io/v1alpha4
+        nodes:
+EOF
+    for i in $(seq 1 $control_planes); do
+        echo "    - role: control-plane" >> /tmp/kind-config.yaml
+    done
+    cat <<EOF >> /tmp/kind-config.yaml
+        - role: worker
+        - role: worker
+        - role: worker
```

Comment:
isn't this part exactly the same as the one from bellow?

Reply:
yes, I will encapsulate this to function.